build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/client_api_module.rb

```diff
@@ -259,7 +260,9 @@ def shape_class_name(shape)
         if @service.protocol == 'api-gateway' && type == 'timestamp'
           shape['timestampFormat'] = 'iso8601'
         end
-        if SHAPE_CLASSES.key?(type)
+        if type == 'structure' && shape['document']
+          ["Shapes::DocumentShape", shape]
+        elsif SHAPE_CLASSES.key?(type)
```

Comment:
Why is this a special case? Also you can use `document_struct?(shape)` here.

Reply:
Good call, `document_struct?(shape)` works here.  This is a special case because document types have "structure" as the type, but do not behave at all like structures - specifically they will never have any members.  This breaks a number of assumptions we have for our `Shapes::StructureShape` and is why I ended up creating a `Shapes::DocumentShape` instead of adding a property onto StructureShape.  All of the other types in `SHAPE_CLASSES` map neatly from type to shape class, but Document doesn't.